compiler/rustc_codegen_ssa/src/mir/block.rs

```diff
@@ -1003,9 +1004,32 @@ impl<'a, 'tcx, Bx: BuilderMethods<'a, 'tcx>> FunctionCx<'a, 'tcx, Bx> {
             (args, None)
         };
 
+        let const_vec_arg_indexes = if let Some(def) = def {
+            let val = bx.tcx().codegen_fn_attrs(def.def_id());
+            let const_vector_indices = &val.const_vector_indices;
+            if let Some(const_vector_i) = const_vector_indices {
+                &const_vector_i.as_ref()
+            } else {
+                &Vec::<usize>::new()
+            }
+        } else {
+            &Vec::<usize>::new()
+        };
+
         let mut copied_constant_arguments = vec![];
         'make_args: for (i, arg) in first_args.iter().enumerate() {
-            let mut op = self.codegen_operand(bx, &arg.node);
+            let mut op = if const_vec_arg_indexes.contains(&i) {
```

Comment:
I think you can just remove this `if`. It's always correct to use this translation if the value happens to be a constant SIMD vector, right?

In fact, should we maybe do this in `OperandRef::from_const`? Then it just happens for all constant SIMD vectors? That would in fact make the attribute entirely optional -- a nice-to-have to prevent LLVM errors, but not actually required to unblock stdarch.